swatch-core/src/main/java/org/candlepin/subscriptions/security/RoleProvider.java

```diff
@@ -42,20 +43,24 @@ public RoleProvider(String rulePrefix, boolean devModeEnabled) {
     this.devModeEnabled = devModeEnabled;
   }
 
-  public List<String> getRoles(Collection<String> permissions) {
-    List<String> roles = new ArrayList<>();
+  public Collection<String> getRoles(Collection<String> permissions) {
     if (permissions == null) {
-      return roles;
+      return List.of();
     }
 
+    Set<String> roles = new HashSet<>();
     // By default, we look for the subscriptions:*:* permission (unless
     // configured otherwise).
-    if (devModeEnabled || permissions.contains(rulePrefix + ":*:*")) {
+    if (permissions.contains(rulePrefix + ":*:*")) {
       roles.add(SWATCH_ADMIN_ROLE);
     }
     if (permissions.contains(rulePrefix + ":reports:read")) {
       roles.add(SWATCH_REPORT_READER);
     }
+    if (devModeEnabled) {
+      roles.add(SWATCH_ADMIN_ROLE);
+      roles.add(ROLE_INTERNAL);
```

Comment:
I think it makes sense to also add the internal role when using DEV_MODE=true to use the internal endpoints (I needed for the conduit service).